tokio/src/net/tcp/stream.rs

```diff
@@ -915,6 +915,97 @@ impl TcpStream {
             .try_io(Interest::WRITABLE, || (&*self.io).write_vectored(bufs))
     }
 
+    /// Try to call a read I/O function, clearing read readiness if `f` returns
+    /// `WouldBlock`
+    ///
+    /// # Return value
+    ///
+    /// This functions returns exactly the same as `f`
+    pub fn try_read_io<R>(&self, f: impl FnMut() -> io::Result<R>) -> io::Result<R> {
+        self.io.registration().try_io(Interest::READABLE, f)
+    }
+
+    /// Polls for read readiness.
+    ///
+    /// If the tcp stream is not currently ready for reading, this method will
+    /// store a clone of the `Waker` from the provided `Context`. When the tcp
+    /// stream becomes ready for reading, `Waker::wake` will be called on the
+    /// waker.
+    ///
+    /// Note that on multiple calls to `poll_read_io`, only the `Waker` from
+    /// the `Context` passed to the most recent call is scheduled to receive a
+    /// wakeup. (However, `poll_write_io` retains a second, independent waker.)
+    ///
+    /// This function is intended for cases where customized I/O operations
+    /// may change the readiness of the underlying socket.
+    /// If the `f` function returns an error `WouldBlock`, then the read
+    /// readiness will be cleared and returns `Poll::Pending`.
```

Comment:
This should either be expanded to say something similar to `try_read_io`, or it should refer to the documentation of `try_read_io`.

Reply:
I have unresolved this. It does not seem like you made any change related to it.

Reply:
Please check the latest commit, I have already fixed the docs for `try_write_io`.

Reply:
Done. Please check if the docs for `poll_read_io` and `poll_write_io` are correct.